flowcraft/generator/engine.py

```diff
@@ -1382,10 +1384,14 @@ def render_pipeline(self):
         # write to file dict_viz
         self.dag_to_file(dict_viz)
 
-        # Write tree forking information for dotfile
-        with open(os.path.join(dirname(self.nf_file),
-                               ".forkTree.json"), "w") as fh:
-            fh.write(json.dumps(self._fork_tree))
+        # Write tree forking information
+        resources_dir = os.path.join(dirname(self.nf_file), "resources")
+
+        if not os.path.exists(resources_dir):
+            os.mkdir(resources_dir)
+        outfile_tree_fork = open(os.path.join(resources_dir, "forkTree.json"), "w")
+        outfile_tree_fork.write(json.dumps(dict_viz))
+        outfile_tree_fork.close()
```

Comment:
Hmm interesting that you are keeping consistency here between both methods. But now you can see that some duplication exists between the two methods? Maybe you could write a function called something like `write_dag_to_file` and use it in both methods.
Then you basically put everything inside that function and re-use in both places:

```python
def write_dag_to_file(file_name, dict_viz):
    resources_dir = os.path.join(dirname(self.nf_file), "resources")
    if not os.path.exists(resources_dir):
            os.mkdir(resources_dir)
    outfile_tree_fork = open(os.path.join(resources_dir, file_name), "w")
    outfile_tree_fork.write(json.dumps(dict_viz))
    outfile_tree_fork.close()
```

or you can even go with `with open`... Then you just call the function in both places. Something like:

```python
write_dag_to_file('forkTree.json', dict_viz)
```